agent/xdsv2/resources_test.go

```diff
@@ -0,0 +1,108 @@
+package xdsv2
+
+import (
+	envoy_cluster_v3 "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	envoy_listener_v3 "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	"github.com/hashicorp/consul/agent/xds/response"
+	"github.com/hashicorp/consul/envoyextensions/xdscommon"
+	proxytracker "github.com/hashicorp/consul/internal/mesh/proxy-tracker"
+	meshv1alpha1 "github.com/hashicorp/consul/proto-public/pbmesh/v1alpha1"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+func TestResources_ImplicitDestinations(t *testing.T) {
+
+	cases := map[string]struct {
+	}{
+		"l4-single-implicit-destination-tproxy": {},
```

Comment:
see my comment below.  when I implemented the AlpnProtocol router match, I totally missed adding in logic for the proxyState -> xds conversion.  so no tests failed and it was not until there was generated xds that you can verify it.  There is no other place to put this kind of test so I added it here.  I did not think we need to recreate all scenarios here since there should be less logic in this step of the conversion.  I don't think we need many tests here, but we need something to able to assert that IR proto changes we make then properly get converted to XDS.  Happy to entertain other ideas!